addons/project/project.py

```diff
@@ -686,6 +686,12 @@ def onchange_user_id(self, cr, uid, ids, user_id, context=None):
             vals['date_start'] = fields.datetime.now()
         return {'value': vals}
 
+    def onchange_date_deadline(
+            self, cr, uid, ids, date_end, date_deadline, context=None):
+        if not date_end or (date_end[:10] == self.browse(
+                cr, uid, ids, context=context).date_deadline):
```

Comment:
So if the dates are already the same, you modify it anyway without the time information? Why?